src/reload.go

```diff
@@ -0,0 +1,373 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type NamespaceRenderingData struct {
+	LeaderVHost string `json:"-" toml:"leader_vhost"`
+	Leader      LeaderController
+	Apps        map[string]App
+	RoutingTag  string `json:"-" toml:"routing_tag"`
+	KnownVHosts Vhosts
+}
+
+type TemplateRenderingData struct {
+	Xproxy              string
+	LeftDelimiter       string                            `json:"-" toml:"left_delimiter"`
+	RightDelimiter      string                            `json:"-" toml:"right_delimiter"`
+	MaxFailsUpstream    *int                              `json:"max_fails,omitempty"`
+	FailTimeoutUpstream string                            `json:"fail_timeout,omitempty"`
+	SlowStartUpstream   string                            `json:"slow_start,omitempty"`
+	Namespaces          map[string]NamespaceRenderingData `json:"namespaces"`
+}
+
+func reload() error {
```

Comment:
Maybe we expose a channel / queue here to receive a signal for reload.

Reply:
Done